### PR TITLE
fix cli dependencies and remove workspaces

### PR DESCRIPTION
**Pull request by jrhizor**

### CODE REVIEW

File: cli/src/helpers/local/configFileGlobal.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import chalk from 'chalk';
 import fs from 'fs';
-import { getTypedStoatConfig, readStoatConfig } from 'stoat-action/src/configHelpers';
-import {StoatConfigSchema} from "stoat-action/src/schemas/stoatConfigSchema";
-import { getTemplate } from 'stoat-action/src/templateHelpers';
-import { Template } from 'stoat-action/src/types';
+import { getTypedStoatConfig, readStoatConfig } from '../../../../action/src/configHelpers';
+import {StoatConfigSchema} from "../../../../action/src/schemas/stoatConfigSchema";
+import { getTemplate } from '../../../../action/src/templateHelpers';
+import { Template } from '../../../../action/src/types';
```
Comment (tuliren) [inline cli/src/helpers/local/configFileGlobal.ts:6]: I am surprised that this works. Are the dependencies messed up? Like if the `templateHelpers` depends on a library that's only in the `action` project, the `cli` project won't have that, and the code will fail.

**Review by tuliren — APPROVED** [review]

File: .github/workflows/test-action.yaml [hunk 2]
```diff
@@ -38,10 +38,11 @@ jobs:
             ${{ github.workspace }}/action/node_modules
           key: ${{ runner.os }}-node-16-action-${{ hashFiles('action/yarn.lock') }}
       - name: Install dependencies
-        working-directory: .
+        working-directory: action
```
Comment (tuliren) [inline .github/workflows/test-action.yaml:41]: This action has a default `working-directory` set on line 22. This line can be removed.

File: .github/workflows/test-action.yaml [hunk 3]
```diff
@@ -38,10 +38,11 @@ jobs:
...
         run: |
           yarn install --frozen-lockfile
       - name: Build and test stoat action
+        working-directory: action
```
Comment (tuliren) [inline .github/workflows/test-action.yaml:45]: Same here.

File: cli/src/helpers/local/configFileGlobal.ts (hunk 1 above)
Comment (tuliren) [reply cli/src/helpers/local/configFileGlobal.ts:6]: I think this is a real issue, based on the latest build failure.

File: cli/package.json [hunk 4]
```diff
@@ -30,29 +30,36 @@
...
     "inquirer": "^9.1.4",
+    "js-yaml": "^4.1.0",
     "latest-version": "^7.0.0",
     "marked": "^4.2.3",
     "open": "^8.4.0",
     "path": "^0.12.7",
     "portfinder": "^1.0.32",
     "prettier": "2.7.1",
-    "stoat-action": "*"
+    "utility-types": "^3.10.0"
```
Comment (tuliren) [inline cli/package.json:55]: `utility-types` should be moved to `devDependencies`.

File: cli/src/helpers/local/configFileGlobal.ts (hunk 1 above)
Comment (jrhizor) [reply cli/src/helpers/local/configFileGlobal.ts:6]: it looks like the "correct" thing is to install the action first

Comment (tuliren) [reply cli/src/helpers/local/configFileGlobal.ts:6]: Nice.

**Pull request merged**